build-tools/packages/build-cli/src/library/release.ts

```diff
@@ -159,10 +178,77 @@ export function toReportKind(
 			break;
 		}
 
+		case "legacy-compat": {
+			for (const [pkg, details] of Object.entries(report)) {
+				toReturn[pkg] = details.ranges.legacyCompat;
+			}
+			break;
+		}
+
 		default: {
 			throw new Error(`Unexpected ReportKind: ${kind}`);
 		}
 	}
 
 	return toReturn;
 }
+
+/**
+ * Generates a new version representing the next version in a legacy compatibility range based on a specified multiple of minor versions.
+ *
+ * @param version - A string representing the current version.
+ * @param interval - The multiple of minor versions to use for calculating the next version in the range.
+ * @param releaseGroup - Release group
+ * @param schemeToUse - The version scheme.
+ *
+ * @returns A string representing the next version in the legacy compatibility range.
+ */
+export function getLegacyCompatVersionRange(
+	version: string,
+	interval: LegacyCompatInterval,
+	releaseGroup: ReleaseGroup | undefined,
+	schemeToUse: VersionScheme,
+): string {
+	for (const [group, intervalValue] of Object.entries(interval.legacyCompatInterval)) {
+		if (group === releaseGroup && intervalValue > 0) {
```

Comment:
Instead of walking group, you can just check for `releaseGroup` in `legacyCompatInterval`.
```typescript
const intervalValue = legacyCompatInterval[releaseGroupOrPackage]
if (intervalValue > 0) {
```
should be okay.
(Hints `releaseGroup` argument should be `releaseGroupOrPackage` to handle non-grouped packages. Commented earlier.)